lib/src/zenohUTransport.cpp

```diff
@@ -22,190 +22,99 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <up-client-zenoh-cpp/message/messageBuilder.h>
-#include <up-client-zenoh-cpp/message/messageParser.h>
 #include <up-client-zenoh-cpp/transport/zenohUTransport.h>
 #include <up-client-zenoh-cpp/session/zenohSessionManager.h>
 #include <up-cpp/uuid/serializer/UuidSerializer.h>
 #include <up-cpp/uri/serializer/LongUriSerializer.h>
+#include <up-cpp/transport/datamodel/UPayload.h>
 #include <spdlog/spdlog.h>
 #include <zenoh.h>
+#include <up-core-api/uattributes.pb.h>
 
 using namespace std;
 using namespace uprotocol::uri;
 using namespace uprotocol::uuid;
 using namespace uprotocol::v1;
+using namespace uprotocol::utransport;
 
-ZenohUTransport& ZenohUTransport::instance(void) noexcept {
+ZenohUTransport::ZenohUTransport() noexcept {
+    /* by default initialized to empty strings */
+    ZenohSessionManagerConfig sessionConfig;
 
-    static ZenohUTransport zenohUtransport;
-
-    return zenohUtransport;
-}
-
-UStatus ZenohUTransport::init() noexcept {
-    
-    UStatus status;
-
-    if (0 == refCount_) {
-
-        std::lock_guard<std::mutex> lock(mutex_);
-
-        if (0 == refCount_) {
-
-            /* by default initialized to empty strings */
-            ZenohSessionManagerConfig sessionConfig;
-
-            if (UCode::OK != ZenohSessionManager::instance().init(sessionConfig)) {
-                spdlog::error("zenohSessionManager::instance().init() failed");
-                status.set_code(UCode::INTERNAL);
-                return status;
-            }
-
-            if (ZenohSessionManager::instance().getSession().has_value()) {
-                session_ = ZenohSessionManager::instance().getSession().value();
-            } else {
-                status.set_code(UCode::INTERNAL);
-                return status;
-            }
-        }
-        termPending_ = false;
-        refCount_.fetch_add(1);
+    if (UCode::OK != ZenohSessionManager::instance().init(sessionConfig)) {
+        spdlog::error("zenohSessionManager::instance().init() failed");
+        uSuccess_.set_code(UCode::INTERNAL);
+    }
 
+    if (ZenohSessionManager::instance().getSession().has_value()) {
+        session_ = ZenohSessionManager::instance().getSession().value();
     } else {
-        refCount_.fetch_add(1);
+        uSuccess_.set_code(UCode::INTERNAL);
```

Comment:
@billpittman  - what is the usage of uSuccess_?

Reply:
uSuccess and rpcSuccess are used to verify that the zenohUTransport and zenohRpcClient classes are properly initialized.  Now that we've moved initialization to the constructors, there is no pass/fail, just constructed or not.  So upClientZenoh::Instance() checks both those variables before allowing the instance to be successfully returned to the caller.

Reply:
but you are not checking them anywhere i think

Reply:
They're checked in upZenohClient::Instance().  The only way that function returns a non-null pointer is if both those status fields are OK.